_pytest/junitxml.py

```diff
@@ -337,6 +339,10 @@ def pytest_runtest_logreport(self, report):
                 reporter = self._opentestcase(report)
                 reporter.append_pass(report)
         elif report.failed:
+            if self.last_report and self.last_report.failed:
```

Comment:
this one will report a double fail if 2 xdist failures of a call happen at the same time

when reports from pytest-xdist come in, the report order is intermixed and cant be used as if it was linear